lib/api/core/validation/index.js

```diff
@@ -1,690 +1,680 @@
+      }
     }
 
-    if (fieldSpec.multivalued.hasOwnProperty('minCount') && fieldSpec.multivalued.hasOwnProperty('maxCount') && fieldSpec.multivalued.minCount > fieldSpec.multivalued.maxCount) {
-      throwOrStoreError(`In ${indexName}.${collectionName}.${fieldName}, "minCount" can not be greater than "maxCount".`, verboseErrors, errors);
+    if (errors.length > 0) {
+      return {errors, isValid: false};
     }
-  }
 
-  if (errors.length > 0) {
-    return {isValid: false, errors: errors};
+    return {isValid: true};
   }
-  return {isValid: true};
-};
 
-/**
- * @param {string} indexName
- * @param {string} collectionName
- * @param {*} validatorFilter
- * @param {boolean} dryRun
- * @returns {Promise}
- */
-Validation.prototype.curateValidatorFilter = function validationCurateValidatorFilter (indexName, collectionName, validatorFilter, dryRun) {
-  var
-    promise,
-    query = {
-      bool: {
-        must: validatorFilter
-      }
-    };
+  /**
+   * @param {string} indexName
+   * @param {string} collectionName
+   * @param {*} validatorFilter
+   * @param {boolean} dryRun
+   * @returns {Promise}
+   */
+  curateValidatorFilter(indexName, collectionName, validatorFilter, dryRun) {
+    const
+      query = {
+        bool: {
+          must: validatorFilter
+        }
+      };
 
-  promise = this.dsl.validate(query);
+    const promise = this.dsl.validate(query);
 
-  if (!dryRun) {
-    promise.then(() => {
-      return this.dsl.register(indexName, collectionName, query);
-    });
+    if (!dryRun) {
+      promise.then(() => this.dsl.register(indexName, collectionName, query));
+    }
+
+    return promise;
   }
 
-  return promise;
-};
+  /**
+   * @param {ValidationType} validationType
+   * @throws {PluginImplementationError}
+   */
+  addType(validationType) {
+    if (!validationType.typeName) {
+      throw new PluginImplementationError('The typeName property must be defined in the validation type object.');
+    }
 
-/**
- * @param {ValidationType} validationType
- * @throws {PluginImplementationError}
- */
-Validation.prototype.addType = function validationAddType (validationType) {
-  if (!validationType.typeName) {
-    throw new PluginImplementationError('The typeName property must be defined in the validation type object.');
-  }
+    if (!validationType.validate || typeof validationType.validate !== 'function') {
+      throw new PluginImplementationError(`The type ${validationType.typeName} must implement the function 'validate'.`);
+    }
 
-  if (!validationType.validate || typeof validationType.validate !== 'function') {
-    throw new PluginImplementationError(`The type ${validationType.typeName} must implement the function 'validate'.`);
-  }
+    if (!validationType.validateFieldSpecification || typeof validationType.validateFieldSpecification !== 'function') {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 120 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)